src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -255,7 +266,7 @@ public JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName, ref ReadSta
                 // First try sorted lookup.
                 int propertyIndex = frame.PropertyIndex;
 
-                // This .Length is consistent no matter what json data intialized _propertyRefsSorted.
+                // Cache the .Length in case it changes by another thread. It may become longer but never shorter (thus IndexOutOfRangeException avoided).
                 int count = _propertyRefsSorted.Length;
```

Comment:
Seems like it'd be safer to store _propertyRefsSorted into a local and then use that local for everything (including accessing its .Length).

Reply:
Without a `volatile` on _propertyRefsSorted, could the jitter optimize away the local?

Reply:
After further thought, adding a local + `volatile` would avoid the need for the `lock`.

Reply:
The new commit uses local + volatile. No perf or stability changes.